docs/sdk2/usage/CUSTOM_OBJECT_SYNC.md

```diff
@@ -0,0 +1,257 @@
+|---|---|---|
+| [CustomObjectDraft](https://docs.commercetools.com/api/projects/custom-objects#customobjectdraft) | `key` |  Also, the custom objects in the target project are expected to have the `key` fields set. | 
+| [CustomObjectDraft](https://docs.commercetools.com/api/projects/custom-objects#customobjectdraft) | `container` |  Also, the custom objects in the target project are expected to have the `container` fields set. | 
+
+####  SyncOptions
+
+After the `projectApiRoot` is set up, a `CustomObjectSyncOptions` should be built as follows:
+````java
+// instantiating a CustomObjectSyncOptions
+final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder.of(projectApiRoot).build();
+````
+
+`SyncOptions` is an object which provides a place for users to add certain configurations to customize the sync process.
+Available configurations:
+
+##### errorCallback
+A callback that is called whenever an error event occurs during the sync process. Each resource executes its own 
+error-callback. When the sync process of a particular resource runs successfully, it is not triggered. It contains the 
+following context about the error-event:
+
+* sync exception
+* custom object draft from the source
+* custom object of the target project (only provided if an existing custom object could be found)
+* a fake list of update actions, as custom objects API does not provide update actions. [NoopResourceUpdateAction.java](#todo)
+
+````java
+ final Logger logger = LoggerFactory.getLogger(CustomObjectSync.class);
+ final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder
+         .of(projectApiRoot)
+         .errorCallback((syncException, draft, customObject, updateActions) -> 
+            logger.error(new SyncException("My customized message"), syncException)).build();
+````
+    
+##### warningCallback
+A callback is called whenever a warning event occurs during the sync process. Each resource executes its own 
+warning-callback. When the sync process of a particular resource runs successfully, it is not triggered. It contains the 
+following context about the warning message:
+
+* sync exception
+* custom object draft from the source 
+* custom object of the target project (only provided if an existing custom object could be found)
+
+````java
+ final Logger logger = LoggerFactory.getLogger(CustomObjectSync.class);
+ final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder
+         .of(projectApiRoot)
+         .warningCallback((syncException, draft, customObject) -> 
+            logger.warn(new SyncException("My customized message"), syncException)).build();
+````
+
+##### beforeUpdateCallback
+In theory, `CustomObjectSyncOptions` provides callback before update operation. User can customize their own callback and inject
+into sync options. However, in the actual case, `beforeUpdateCallback`is not triggered in the custom object sync process. When
```

Comment:
Maybe give a link to the docs that there are no updates available for custom objects: https://docs.commercetools.com/api/projects/custom-objects#create-or-update-customobject